crates/viewer/re_view_spatial/src/eye.rs

```diff
@@ -150,235 +160,229 @@ impl Eye {
-///
-/// This is either a first person camera or an orbital camera,
-/// controlled by [`Eye3DKind`].
-/// We combine these two modes in one struct because they share a lot of state and logic.
-///
-/// Note: we use "eye" so we don't confuse this with logged camera.
-#[derive(Clone, Copy, Debug, PartialEq)]
-pub struct ViewEye {
-    /// First person or orbital?
-    kind: Eye3DKind,
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct LastTrackedEntity {
+    /// Currently tracked entity.
+    pub entity: EntityPath,
+}
 
-    /// Center of orbit, or camera position in first person kind.
-    center: Vec3,
+#[derive(Clone, Debug, PartialEq)]
+struct EyeInterpolation {
+    elapsed_time: f32,
+    start: Eye,
+}
 
-    /// Ignored for [`Eye3DKind::FirstPerson`],
-    /// but kept for if/when the user switches to orbital kind.
-    orbit_radius: f32,
+impl EyeInterpolation {
+    pub fn target_time(start: &Eye, stop: &Eye) -> Option<f32> {
+        // Take more time if the rotation is big:
+        let angle_difference = start
+            .world_from_rub_view
+            .rotation()
+            .angle_between(stop.world_from_rub_view.rotation());
 
-    /// Rotate to world-space from view-space (RUB).
-    world_from_view_rot: Quat,
+        // Threshold to avoid doing pointless interpolations that trigger frame requests.
+        let distance = start.pos_in_world().distance(stop.pos_in_world());
+        if angle_difference < 0.01 && distance < 0.0001 {
+            None
+        } else {
+            Some(egui::remap_clamp(
+                angle_difference,
+                0.0..=std::f32::consts::PI,
+                0.2..=0.7,
+            ))
+        }
+    }
+}
 
+/// Some non-persistent state for the eye.
+///
+/// Note: we use "eye" so we don't confuse this with logged camera.
+#[derive(Default, Clone, Debug, PartialEq)]
+pub struct EyeState {
     /// Vertical field of view in radians.
-    fov_y: f32,
+    fov_y: Option<f32>,
 
-    /// The up-axis of the eye itself, in world-space.
+    velocity: Vec3,
+
+    /// The lasst tracked entity.
     ///
-    /// Initially, the up-axis of the eye will be the same as the up-axis of the scene (or +Z if
-    /// the scene has no up axis defined).
-    /// Rolling the camera (e.g. middle-click) will permanently modify the eye's up axis, until the
-    /// next reset.
+    /// This should not be used to get the current tracked entity, get that
+    /// via view properties instead.
+    pub last_tracked_entity: Option<LastTrackedEntity>,
+
+    interpolation: Option<EyeInterpolation>,
+
+    /// How many radians the camera has spun.
+    spin: Option<f64>,
+
+    pub last_eye: Option<Eye>,
+
+    /// The time this was last interacted with in egui time.
     ///
-    /// A value of `Vec3::ZERO` is valid and will result in 3 degrees of freedom, although we never
-    /// use it at the moment.
+    /// None: Hasn't been interacted with yet.
+    pub last_interaction: Option<f64>,
```

Comment:
maybe?

```suggestion
    pub last_interaction_time: Option<f64>,
```

the fact that this doesn't have a dimension like `Instant` is a bit annoying. but using egui time here makes sense